code/network/lib/Loot/Network/ZMQ/Client.hs

```diff
@@ -558,92 +468,72 @@ runBroker = do
 
             -- Getting internal identifier of the node provided (if provided)
-            iIdM <- runMaybeT $ do
-                nodeId <- MaybeT $ pure nodeIdM
-                MaybeT $ HMap.lookup nodeId <$>
-                    atomically (readTVar (ztPeers ^. pivConnected))
-
-            if isJust nodeIdM && isNothing iIdM
-            then ztCliLog Warning $ "Coludn't send to peer, can't resolve: " <> show nodeIdM
-            else do
-                -- At this moment either both are nothing or both are just.
-
-                -- Choosing at random if not provided (both are nothing).
-                (iIdFinal :: Maybe ZTInternalId) <-
-                    maybe (fmap snd <$> choosePeer) (pure . Just) iIdM
-
-
-                -- If it's nothing it means that we couldn't choose peer
-                -- at random -- the peer list is empty, so we just silently skip.
-                whenJust iIdFinal $ \(unZTInternalId -> iId) ->
-                    Z.sendMulti ztCliBack $ NE.fromList $
-                    [iId, "", tag_normal, unMsgType msgT] ++ msg
-
-    let routerToClients (ZTInternalId -> iId) (MsgType -> msgT) msg =
-          atomically (HMap.lookup iId <$> peersRevMap ztPeers) >>= \case
-            Nothing ->
-                ztCliLog Warning $ "routerToClients: couldn't resolve the node"
-            Just nodeId -> do
-                onHeartbeat nodeId
-                clientIdM <- atomically $ Map.lookup msgT <$> readTVar ztMsgTypes
-                maybe (ztCliLog Warning $ "routerToClients: couldn't find client " <>
-                                          "with this message type")
-                      (\clientId ->
-                        sendToClient clientId (nodeId, Response msgT msg))
-                      clientIdM
-
-    let subToClients (Subscription -> subK) (ZTInternalId -> iId) content =
-            atomically (HMap.lookup iId <$> peersRevMap ztPeers) >>= \case
-              Just nodeId -> do
-                  onHeartbeat nodeId
-                  unless (subK == heartbeatSubscription) $ do
-                      cids <-
-                          atomically $
-                          fromMaybe mempty . Map.lookup subK <$>
-                          readTVar ztSubscriptions
-                      forM_ cids $ \clientId ->
-                          sendToClient clientId (nodeId, Update subK content)
-              Nothing -> ztCliLog Warning $ "subToClients: can't resolve iId" <> show iId
+            nodeId' <- maybe choosePeer (pure . Just) nodeIdM
+            whenJust nodeId' $ \nodeId ->
+                atomically (HMap.lookup nodeId <$> readTVar ztPeers) >>= \case
+                    Nothing ->
+                        ztCliLog Warning $
+                        "Coludn't send to peer, can't resolve: " <> show nodeId
+                    Just PeerRes{ prBack } -> do
+                        Z.sendMulti prBack $ NE.fromList $
+                            ["", unMsgType msgT] ++ msg
```

Comment:
is this `""` really needed to send?

Reply:
it seems that server also expects `clientId` as first frame, doesn't it?

Reply:
This is `DEALER`, so yes, the first frame must be `""`, read the guide.

Reply:
>This is DEALER, so yes, the first frame must be "", read the guide.

ok, I will. but aside the guide: which motivation?